lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -714,6 +697,101 @@ llvm::SmallVector<int64_t> BufferAttr::getShape() const {
   return bufferShape;
 }
 
+static mlir::AffineMap createL1Map(::mlir::MLIRContext *context,
+                                   GridAttr workerGrid,
+                                   SystemDescAttr systemDesc,
+                                   ::llvm::ArrayRef<unsigned> chipIds) {
+  mlir::AffineMap workerMap = workerGrid.getMapping();
+  mlir::SmallVector<mlir::AffineExpr> l1MapResults(workerMap.getNumDims());
+  mlir::AffineExpr shardIndexing = getAffineConstantExpr(0, context);
+  mlir::AffineExpr shardVolumeExpr = getAffineConstantExpr(1, context);
+
+  // Compute the projection of the layout onto its own logical grid.
+  // Simultaneously compute the indexing of shards within each core.
+  for (int i = workerMap.getNumDims() - 1; i >= 0; i--) {
+    mlir::AffineExpr linearIdx = getAffineDimExpr(i, context);
+    mlir::AffineExpr shardDim = getAffineSymbolExpr(i, context);
+    l1MapResults[i] = linearIdx.floorDiv(shardDim);
+    shardIndexing = (linearIdx % shardDim) * shardVolumeExpr + shardIndexing;
+    shardVolumeExpr = shardVolumeExpr * shardDim;
+  }
+
+  // Compose the logical grid projection with the device grid mapping, now we
+  // have a projection onto the physical grid.
+  mlir::AffineMap gridProjection = workerMap.compose(mlir::AffineMap::get(
+      workerMap.getNumDims(), workerMap.getNumDims(), l1MapResults, context));
+
+  // Finally we append the indexing of shards within each core.
+  mlir::SmallVector<mlir::AffineExpr> l1Map(gridProjection.getResults());
+  l1Map.push_back(shardIndexing);
+  return mlir::AffineMap::get(workerMap.getNumDims(), workerMap.getNumDims(),
+                              l1Map, context);
+}
+
+static mlir::AffineMap
+createDramMap(::mlir::MLIRContext *context, GridAttr workerGrid, ArchAttr arch,
```

Comment:
if you could just add comment for both functions createL1Map, createDramMap how map looks like for an example grid.